pgml-extension/src/bindings/transformers/transformers.py

```diff
@@ -110,6 +112,31 @@ def __call__(self, inputs, **kwargs):
         return outputs
 
 
+class ThreadedGeneratorIterator:
+    def __init__(self, output):
+        self.done_data = []
+        self.output = output
+        self.done = False
+
+        def do_work(g):
+            for x in g.output:
+                g.done_data.append(x)
+            g.done = True
+        thread = Thread(target=do_work, args=(self,))
+        thread.start()
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if len(self.done_data) > 0:
+            return self.done_data.pop(0)
+        elif self.done:
+            raise StopIteration
+        time.sleep(0.1)
```

Comment:
Why does this need to sleep?

Reply:
If I don't have it sleep I get:
```ERROR:  RecursionError: maximum recursion depth exceeded```

Reply:
That's weird...

Reply:
This is because we hit the recursion depth before a new token is added to done_data. I'd suggest moving to a while loop rather than recursing since Python is a ... does not support tail recursion.

Reply:
After moving to a while loop, I think it's worth a big fat comment warning that we are spin locking in this thread, and sleeping for some amount of time on each iteration in a pure `else` case, and sleeping for some amount of time may be a better cost/latency tradeoff.

Reply:
Is it possible to lock/wait/notify with postgres in this case?

Reply:
https://docs.python.org/3.10/library/queue.html

https://stackoverflow.com/questions/14508906/sending-messages-between-class-threads-python